pkg/network/vmispec/network.go

```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package vmispec
+
+import v1 "kubevirt.io/api/core/v1"
+
+func lookupPodNetwork(networks []v1.Network) *v1.Network {
```

Comment:
@EdDev, @iholder-redhat  what do you think about this implementation?
```go
func lookupPodNetwork(networks []v1.Network) *v1.Network {
	for i, network := range networks {
		if network.Pod != nil {
			return &networks[i]
		}
	}
	return nil
}
```

Reply:
This is the alternative to the copy I did. In this case, I do not see any benefit... both have pros and cons.
(this original one, which uses a copy, returns a real copy, so if the caller modifies it, the original data is not affected. This can be a good or bad thing, depending on what the caller wants to do)

If I am unsure, I usually pick the immutable approach as it is safer across potential usages. But I do not have any preference at this moment on this.

Reply:
@EdDev, Thanks for your response.

Reply:
If performance is negligible and we don't need to return a pointer (i.e. in order for other component to modify the same object) I would vote for changing the function to not return a pointer.

WDYT of this:
```
func lookupPodNetwork(networks []v1.Network) v1.Network {
	for i, network := range networks {
		if network.Pod != nil {
			return &networks[i]
		}
	}
	return nil
}
```

Same thing as @orelmisan suggested, but returning by-value.

BTW - I would say that on the default case, unless there's a reason, it's better to return by-value as it's safer and because of compiler optimizations not always slower.

Reply:
Hmm.. returning by value in this case will require another boolean "exists" to be returned, as we cannot return a nil anymore. Is this what you prefer?

Reply:
Yup, valid point..
Well - every approach has its pros and cons, I'm okay with what you decide :)